src/main.tsx

```diff
@@ -1,21 +1,24 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App.tsx'
+import React from "react";
+import ReactDOM from "react-dom/client";
+import App from "./App.tsx";
 import "bootstrap/dist/js/bootstrap.bundle.min.js";
 import "bootstrap/dist/css/bootstrap.min.css";
 import "@fortawesome/fontawesome-free/css/all.min.css";
 import "react-toastify/dist/ReactToastify.css";
-import './index.css';
-import { AuthContextProvider } from './Context/AuthContext.tsx';
-import { ToastProvider } from './Context/ToastContext.tsx';
+import "./index.css";
+import { AuthContextProvider } from "./Context/AuthContext.tsx";
+import { ToastProvider } from "./Context/ToastContext.tsx";
 import { ToastContainer } from "react-toastify";
-ReactDOM.createRoot(document.getElementById('root')!).render(
-  <React.StrictMode>
-     <AuthContextProvider>
-      <ToastProvider>
-         <App />
-      </ToastProvider>
-      <ToastContainer />
-    </AuthContextProvider>
-  </React.StrictMode>,
-)
+import BaseUrlContextProvider from "./Context/BaseUrlContext.tsx";
+ReactDOM.createRoot(document.getElementById("root")!).render(
+	<React.StrictMode>
+		<AuthContextProvider>
+			<ToastProvider>
```

Comment:
What is the difference between this approach and having a `ToastContainer` at a top level component and then use `toast.error` and `toast.success`  functions given from the package ?

Reply:
what about declearing a function in a utils file depends on the type of toast and fetching the message from api response ? instead of calling toast.success and toast.error , what if we need to use toast.warning/ info  ?

Reply:
We can indeed make utils for api calls, but for the toasters it is already handled by the package.
It provides us with all types of toast: success, error, warning and info and so on.
![image](https://github.com/OmarBazeed/project-mangement-system/assets/64327685/84cd2170-b3e6-4c95-b189-912265532808)

Reply:
we updated it